### DISCUSSION TITLE
Labo 2 - Recherche

### DISCUSSION THREAD

**Discussion by Greenkiller21** [amt-classroom/amt-classroom.github.io#13] [Q&A]
Lorsque l'on fait une recherche (de client ou de flim), que se passe-il quand on selectionne une ligne (avec le bouton select) et que, ensuite, on modifie notre recherche ? La ligne est-elle toujours selectionnée ou alors on repart d'un état ou rien n'est sélectionné.
Que se passe-il si on selectionne plusieurs lignes ? Doit-on bloquer ceci ou non ?
Autre question, quand on recherche "brooklyn" puis qu'on efface cette recherche, que doit-il se passer ? Doit on limiter les resultats reçus de la base de données ou alors doit-on juste rien afficher ?
Merci

**Comment by Perdjesk** [comment]
> Lorsque l'on fait une recherche (de client ou de flim), que se passe-il quand on selectionne une ligne (avec le bouton select) et que, ensuite, on modifie notre recherche ? La ligne est-elle toujours selectionnée ou alors on repart d'un état ou rien n'est sélectionné.

Dans la version d'exemple, lors d'un changement de la recherche le tableau est remplacé par la liste des résultats, et toute sélection est abandonnée. Ceci était la version la plus simple et directe, plutôt que de par exemple, changer les résultats uniquement après la désélection d'un élément. 
Toutefois si vous pensez qu'une version différente est plus facile ou pertinente vous avez un degré de liberté tant que les fonctionnalités tel que décrites dans l'énoncé sont présentent.

> Que se passe-il si on selectionne plusieurs lignes ? Doit-on bloquer ceci ou non ? 

La version d'exemple ne permet pas de sélectionner plusieurs lignes et force de rentrer chaque location une après l'autre. 

Une version qui permet de sélectionner plusieurs élément de l'inventaire pour un client sélectionné pourrait faire du sens et être utile, mais n'est pas requise. Par contre la multi-sélection des clients ne me semble pas permettre une fonctionnalité supplémentaire. Dans tous les cas l'implémentation serveur se doit d'autoriser uniquement des locations valide.

> Autre question, quand on recherche "brooklyn" puis qu'on efface cette recherche, que doit-il se passer ? Doit on limiter les resultats reçus de la base de données ou alors doit-on juste rien afficher ?

Si le champ de recherche est vide, il n'y a pas de recherche et donc pas de résultats.
